CMakeLists.txt

```diff
@@ -385,6 +385,8 @@ set(SUNSHINE_TARGET_FILES
 
 set_source_files_properties(src/upnp.cpp PROPERTIES COMPILE_FLAGS -Wno-pedantic)
 
+set_source_files_properties(third-party/nanors/rs.c PROPERTIES COMPILE_FLAGS "-include deps/obl/autoshim.h")
```

Comment:
not sure how to turn on auto vectorization in MSVC but with `gcc`/`clang`
adding `-ftree-vectorize` gives you simd accelerated xor for [this loop](https://github.com/sleepybishop/nanors/blob/master/deps/obl/oblas_lite.c#L232). I didn't use intrinsics explicitly for xor since the I let the compiler handle it.

Reply:
Are you suggesting `-ftree-vectorize` instead of including `autoshim.h` or in addition to?

Reply:
in addition to, the autoshim will just choose the proper intrinsic code for the platform, auto vectorization needs to be enabled for faster xor's.

after a bit of reading I think msvc will auto vectorize with `/02` which is the default for release builds, so maybe no changes required at least for windows builds

Reply:
Our Windows builds are using MinGW, so `-ftree-vectorize` should be correct across the board.

Reply:
hmm, didn't seem to work, 

```
objdump -S ./CMakeFiles/sunshine.dir/third-party/nanors/rs.c.o|less
```

```
 18d:   4c 8b 65 d8             mov    -0x28(%rbp),%r12
        for (; ap < ae; ap++, bp++)
 191:   eb 14                   jmp    1a7 <obl_axpy+0x4c>
            *ap ^= *bp;
 193:   0f b6 13                movzbl (%rbx),%edx
 196:   41 0f b6 04 24          movzbl (%r12),%eax
 19b:   31 d0                   xor    %edx,%eax
 19d:   88 03                   mov    %al,(%rbx)
        for (; ap < ae; ap++, bp++)
```

ideally you want at least `pxor` being used, with `-include deps/obl/autoshim.h -ftree-vectorize -msse4 -O2` you should get something like
```
12d:   00 00 00 
           *ap ^= *bp;
130:   f3 0f 6f 04 07          movdqu (%rdi,%rax,1),%xmm0
135:   f3 0f 6f 14 06          movdqu (%rsi,%rax,1),%xmm2
13a:   66 0f ef c2             pxor   %xmm2,%xmm0
13e:   0f 11 04 07             movups %xmm0,(%rdi,%rax,1)
       for (; ap < ae; ap++, bp++)

```

Reply:
Yep, the issue is the optimization flags were broken for C code unless you manually specified an optimized `CMAKE_BUILD_TYPE`. I have a fix in #777.